### ISSUE TITLE
Bug: deleteSQL: Did not find column names in the WHERE Statement

### ISSUE THREAD

**Issue by Sampath-Lokuge** [capacitor-community/sqlite#470]
Hi,

This is an Ionic/Angular native mobile app.

```
Error: Uncaught (in promise): Error: Run: DeleteSQL deleteSQL: Did not find column names in the WHERE Statement
Error: Run: DeleteSQL deleteSQL: Did not find column names in the WHERE Statement
    at returnResult (VM1460 home:770:32)
    at win.androidBridge.onmessage (VM1460 home:745:21)
    at resolvePromise (zone.umd.js:1250:35)
    at zone.umd.js:1157:21
    at zone.umd.js:1173:37
    at _ZoneDelegate.invoke (zone.umd.js:411:30)
    at Object.onInvoke (core.mjs:10981:33)
    at _ZoneDelegate.invoke (zone.umd.js:410:56)
    at Zone.run (zone.umd.js:165:47)
    at zone.umd.js:1314:38
    at _ZoneDelegate.invokeTask (zone.umd.js:445:35)
    at core.mjs:10655:55
```

package.json

{
  "name": "",
  "version": "1.5.23",
  "author": "",
  "homepage": "",
  "scripts": {
    "ng": "ng",
    "start": "ng serve",
    "build": "ng build --prod",
    "lint": "ng lint"
  },
  "private": true,
  "dependencies": {
    "@angular/cdk": "16.2.3",
    "@angular/common": "16.2.4",
    "@angular/core": "16.2.4",
    "@angular/forms": "16.2.4",
    "@angular/platform-browser": "16.2.4",
    "@angular/platform-browser-dynamic": "16.2.4",
    "@angular/router": "16.2.4",
    "@capacitor-community/sqlite": "5.2.3",
    "@capacitor-firebase/authentication": "5.1.0",
    "@capacitor/android": "5.2.3",
    "@capacitor/app": "5.0.0",
    "@capacitor/camera": "5.0.0",
    "@capacitor/core": "5.2.3",
    "@capacitor/haptics": "5.0.0",
    "@capacitor/ios": "5.2.3",
    "@capacitor/keyboard": "5.0.0",
    "@capacitor/network": "5.0.0",
    "@capacitor/preferences": "5.0.0",
    "@capacitor/splash-screen": "5.0.0",
    "@capacitor/status-bar": "5.0.0",
    "@ionic/angular": "7.3.4",
    "@sentry/angular-ivy": "7.54.0",
    "@sentry/capacitor": "0.12.1",
    "@types/lodash": "4.14.178",
    "dayjs": "1.10.7",
    "firebase": "9.6.10",
    "guid-typescript": "1.0.9",
    "lodash": "4.17.21",
    "rxjs": "7.5.0",
    "tslib": "2.2.0",
    "zone.js": "0.13.3"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "16.2.2",
    "@angular-eslint/builder": "16.1.2",
    "@angular-eslint/eslint-plugin": "16.1.2",
    "@angular-eslint/eslint-plugin-template": "16.1.2",
    "@angular-eslint/template-parser": "16.1.2",
    "@angular/cli": "16.2.2",
    "@angular/compiler": "16.2.4",
    "@angular/compiler-cli": "16.2.4",
    "@angular/language-service": "16.2.4",
    "@capacitor/cli": "5.0.0",
    "@ionic/angular-toolkit": "9.0.0",
    "@sentry/cli": "2.20.7",
    "@types/node": "12.11.1",
    "@typescript-eslint/eslint-plugin": "6.7.0",
    "@typescript-eslint/parser": "6.7.0",
    "eslint": "8.49.0",
    "eslint-plugin-import": "2.28.1",
    "eslint-plugin-jsdoc": "46.6.0",
    "eslint-plugin-prefer-arrow": "1.2.3",
    "prettier": "2.5.1",
    "ts-node": "8.3.0",
    "typescript": "4.9.5"
  },
  "description": ""
}

Here it gives an error. But it worked before the Capacitor 5 upgrade.

```
 private deleteExtrasByFareId(fareId: string): Promise<capSQLiteChanges> {
    const sqlQuery = `DELETE FROM Extras WHERE FareId=?`;

    const sqlValues = [fareId];

    return this.sqliteDbConnection.run(sqlQuery, sqlValues, false);
  }
```
Please let me know the issue here. Thanks!

**Comment by jepiqueau** [comment]
@Sampath-Lokuge you should share your code with whzt you give it is difficult to understand. Are you sure that the error comes from this delete statement

**Comment by Sampath-Lokuge** [comment]
I have modified the above code to see the error and yes it comes from the Delete row statement.

**Video:** [SQLite Delete Bug](https://drive.google.com/file/d/1z4RNDoEI9M35oFWBI5vn7G2QBExq-OmG/view?usp=sharing)

```
 private async deleteExtrasByFareId(fareId: string): Promise<void> {
    const sqlQuery1 = `SELECT * FROM Extras`;

    const extras = await this.sqliteDbConnection.query(sqlQuery1);

    console.log(`extras`, extras);

    const sqlQuery = `DELETE FROM Extras WHERE FareId=?;`;

    const sqlValues = [fareId];

    try {
      await this.sqliteDbConnection.run(sqlQuery, sqlValues, false);
    } catch (e) {
      console.log(`SQLite-error`, e);
    }
  }
```

**Comment by jepiqueau** [comment]
@Sampath-Lokuge Are you using API <=13 or higher? You must share your code the difference between 5.0.0 and 5.0.7 is the support of RETURNING which had an impact on the Delete command

**Comment by jepiqueau** [comment]
@Sampath-Lokuge Did you test it on iOS and/or Electron?

**Comment by Sampath-Lokuge** [comment]
> Are you using API <=13 or higher? 

Android 13.

> You must share your code the difference between 5.0.0 and 5.0.7 is the support of RETURNING which had an impact on the Delete command

Same code on both versions.

```
 private deleteExtrasByFareId(fareId: string): Promise<capSQLiteChanges> {
    const sqlQuery = `DELETE FROM Extras WHERE FareId=?`;

    const sqlValues = [fareId];

    return this.sqliteDbConnection.run(sqlQuery, sqlValues, false);
  }
```

> Did you test it on iOS and/or Electron?

No. Only on Android 13 device.

**Comment by jepiqueau** [comment]
@Sampath-Lokuge when you defined 
```
FOREIGN KEY(FareId) REFERENCES Fares(FareId)
```
you do not define an `ON DELETE` mode

**Comment by jepiqueau** [comment]
@Sampath-Lokuge Can you send me the Fares table schema

**Comment by Sampath-Lokuge** [comment]
> you do not define an ON DELETE mode

I have never done it with my app. But it worked in the past. So is that a new requirement? Can I have a link for a doc that you have mentioned about this change?

> Can you send me the Fares table schema

createFaresTable(): Promise<capSQLiteChanges> {
    const query = `
          CREATE TABLE IF NOT EXISTS Fares (
            FareId TEXT PRIMARY KEY NOT NULL,
            Amount REAL NULL,
            Date INTEGER NULL,
            FareTips REAL NULL,
            PaymentTypeId TEXT NULL,
            FareTemplateId TEXT NULL,
            CreatedDate INTEGER NOT NULL,
            UserId TEXT NOT NULL,
            last_modified INTEGER DEFAULT (strftime('%s', 'now')),
            IsExportedAccounts INTEGER DEFAULT 0,
            sql_deleted BOOLEAN DEFAULT 0 CHECK (sql_deleted IN (0, 1)),
            FareTypeId TEXT NULL,
            FOREIGN KEY(PaymentTypeId) REFERENCES PaymentTypes(PaymentTypeId),
            FOREIGN KEY(FareTemplateId) REFERENCES FareTemplates(FareTemplateId),
            FOREIGN KEY(FareTypeId) REFERENCES FareTypes(FareTypeId)
           );
           
          CREATE TRIGGER IF NOT EXISTS Fares_trigger_last_modified
          AFTER UPDATE ON Fares
          FOR EACH ROW WHEN NEW.last_modified < OLD.last_modified
          BEGIN
            UPDATE Fares SET last_modified= (strftime('%s', 'now')) 
            WHERE FareId=OLD.FareId;
          END;
           `;

    return this.sqliteDbConnection.execute(query);
  }

**Comment by jepiqueau** [comment]
@Sampath-Lokuge if not specify it will take the ON DELETE SET DEFAULT behavior.
The problem has been identified and will be fixed in the next release

**Comment by Sampath-Lokuge** [comment]
Sure, Please keep open this ticket till then. Thanks!

**Comment by jepiqueau** [comment]
@Sampath-Lokuge fix in 5.2.5 so i close the issue feel free to re-open it if requires
